packages/rulesets/src/spectral/test/resource-names-restriction.test.ts

```diff
@@ -93,3 +93,427 @@ test("ResourceNameRestriction should find no errors", () => {
     expect(results.length).toBe(0)
   })
 })
+
+test("ResourceNameRestriction should find errors for multiple methods and undefined parameters section", () => {
```

Comment:
I think this huge new test is unnecessary. The only thing that needs to verified is that the function does not crash for an operation with no `parameters`.